cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/JsonMenu/utilities.ts

```diff
@@ -27,65 +22,61 @@ function getJSONConfig(widgetJSONData) {
   } = widgetJSONData;
 
   const configurationGroupsData = configurationGroups.map((groupID: string) => {
-    const groupLabel = groupToInfo[groupID].label;
-    const widgetData = groupToWidgets[groupID].map((widgetID: string) => {
-      const info: IWidgetInfo = widgetInfo[widgetID];
-      const widgetAttributes = widgetToAttributes[widgetID];
-
-      return {
-        'widget-type': info.widgetType,
-        label: info.label,
-        name: info.name,
-        ...(info.widgetCategory && { 'widget-category': info.widgetCategory }),
+    const groupLabel = groupToInfo.get(groupID).get('label');
+    const widgetData = groupToWidgets.get(groupID).map((widgetID: string) => {
+      const info = widgetInfo.get(widgetID);
+      const widgetAttributes = widgetToAttributes.get(widgetID);
+
+      return fromJS({
+        'widget-type': info.get('widgetType'),
+        label: info.get('label'),
+        name: info.get('name'),
+        ...(info.get('widgetCategory') && {
+          'widget-category': info.get('widgetCategory'),
+        }),
         ...(widgetAttributes &&
-          Object.keys(widgetAttributes).length > 0 && {
+          List(widgetAttributes.keys()).size > 0 && {
             'widget-attributes': widgetAttributes,
           }),
-      };
+      });
     });
-    return {
+    return fromJS({
       label: groupLabel,
       properties: widgetData,
-    };
+    });
   });
 
   const filtersData = filters.map((filterID) => {
-    const filterToShowListData = filterToShowList[filterID].map((showID) => {
-      let showListData;
-      if (showToInfo[showID].type) {
-        showListData = {
-          name: showToInfo[showID].name,
-          type: showToInfo[showID].type,
-        };
-      } else {
-        showListData = {
-          name: showToInfo[showID].name,
-        };
-      }
-      return showListData;
+    const filterToShowListData = filterToShowList.get(filterID).map((showID) => {
+      return fromJS({
+        name: showToInfo.get(showID).get('name'),
+        ...(showToInfo.get(showID).get('type') && {
+          type: showToInfo.get(showID).get('type'),
+        }),
+      });
     });
-    return {
-      name: filterToName[filterID],
-      condition: filterToCondition[filterID],
+    return fromJS({
+      name: filterToName.get(filterID),
+      condition: filterToCondition.get(filterID),
       show: filterToShowListData,
-    };
+    });
   });
 
   const config = {
     metadata: {
-      'spec-version': SPEC_VERSION,
+      'spec-version': '1.5',
```

Comment:
i think you should keep the spec version on the variable

Reply:
Resolved